electron-core/src/electron/CapacitorElectronApp.ts

```diff
@@ -268,7 +268,9 @@ export class CapacitorElectronApp {
       ) {
         this.splashScreenReference.hide();
       } else {
-        this.mainWindowReference.show();
+        if (browserWindowConfig.show) {
```

Comment:
Change this to: `if(!this.config.mainWindow.windowOptions.show === false) {`

This allows to check if the show option is explicitly set to false instead of default setting. This should give you the functionality you are looking for without many changes.